src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1816,24 +1961,53 @@ private static BigInteger Subtract(uint[]? leftBits, int leftSign, uint[]? right
                 return Zero;
             }
 
-            if (left._bits == null && right._bits == null)
+            if (left._bits is null && right._bits is null)
             {
                 return left._sign & right._sign;
             }
 
-            uint[] x = left.ToUInt32Array();
-            uint[] y = right.ToUInt32Array();
-            uint[] z = new uint[Math.Max(x.Length, y.Length)];
             uint xExtend = (left._sign < 0) ? uint.MaxValue : 0;
             uint yExtend = (right._sign < 0) ? uint.MaxValue : 0;
 
+            uint[]? leftBufferFromPool = null;
+            int size = (left._bits?.Length ?? 1) + 1;
+            Span<uint> x = size <= StackAllocThreshold ?
+                           stackalloc uint[size]
+                           : leftBufferFromPool = ArrayPool<uint>.Shared.Rent(size);
+            x = x.Slice(0, left.CopyTo(x));
```

Comment:
Does this work?

Reply:
Yes, why not? All tests passed. `Clear` is not necessary here.

Reply:
This shouldn't compile. The second argument for `Slice` should be an `int` (the length) and not `void` from `CopyTo`. Or do I miss something right now?

Reply:
`left` is an instance of `BigInteger`, not of `Span`. `CopyTo` is a new name for old `ToUInt32Array` method.

Reply:
Now I got you. Maybe there's a better name than `CopyTo` as it's really misleading in the context of spans.

Reply:
Hmm, it's hard :smile: because the method do the copy and return the number of copied elements. I can try to choose something close to its semantics.

Reply:
`WriteTo` maybe?

Reply:
👍 
This would read "natural".
```diff
- left.CopyTo(x)
+ left.WriteTo(x)
```